File: frontend/src/types/stats.ts
export interface UserWithStats {
  id: number
  email: string
  username: string
  created_at: string
  total_groups: number
  created_groups: number
  total_reviews: number
  albums_added: number
  has_spotify: boolean
}

export interface UserGuessStatsResponse {
  user_id: number
  group_id: number
  total_guesses: number
  correct_guesses: number
  accuracy: number
}

export interface AlbumReviewStatsResponse {
  album_id: number
  review_count: number
  avg_rating: number | null
  min_rating: number | null
  max_rating: number | null
}

export interface MemberGuessResult {
  guessing_user_id: number
  guessing_username: string
  guessed_user_id: number | null
  guessed_username: string | null
  is_chaos: boolean
  correct: boolean
}

export interface AlbumGuessStatsResponse {
  group_album_id: number
  nominator_user_id: number
  nominator_username: string
  total_guesses: number
  correct_guesses: number
  guesses: MemberGuessResult[]
}

File: backend/app/schemas/stats.py
"""Schemas for group and album statistics."""

from pydantic import BaseModel


class UserGuessStatsResponse(BaseModel):
    user_id: int
    group_id: int
    total_guesses: int
    correct_guesses: int
    accuracy: float


class MemberGuessResult(BaseModel):
    guessing_user_id: int
    guessing_username: str
    guessed_user_id: int | None
    guessed_username: str | None
    is_chaos: bool
    correct: bool


class AlbumGuessStatsResponse(BaseModel):
    group_album_id: int
    nominator_user_id: int | None
    nominator_username: str | None
    total_guesses: int
    correct_guesses: int
    guesses: list[MemberGuessResult]
    # False when the viewer has yet to guess: nominator and per-member results
    # are withheld so the breakdown cannot be used to cheat.
    revealed: bool = True


class AlbumReviewStatsResponse(BaseModel):
    album_id: int
    review_count: int
    avg_rating: float | None
    min_rating: float | None
    max_rating: float | None
